core/api/transport/impl/http_session.cpp

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "http_session.hpp"
+
+#include <boost/config.hpp>
+#include <iostream>
+
+namespace kagome::api {
+
+  /**
+   * @brief process http request, compose and execute response
+   * @tparam Body request body type
+   * @tparam Allocator allocator type
+   * @param req request
+   * @param send sender function
+   */
+  template <class Body>
+  void HttpSession::handleRequest(boost::beast::http::request<Body> &&req) {
+    // Returns a bad request response
+    auto const bad_http_request = [&req](std::string_view message) {
+      boost::beast::http::response<boost::beast::http::string_body> res{
+          boost::beast::http::status::bad_request, req.version()};
+      res.set(boost::beast::http::field::server, kServerName);
+      res.set(boost::beast::http::field::content_type, "text/html");
+      res.keep_alive(req.keep_alive());
+      res.body() = message;
+      res.prepare_payload();
+      return res;
+    };
+
+    // allow only POST method
+    if (req.method() != boost::beast::http::verb::post) {
+      return asyncWrite(bad_http_request("Unsupported HTTP-method"));
+    }
+
+    processRequest(req.body(), shared_from_this());
+  }
+
+  HttpSession::HttpSession(boost::asio::ip::tcp::socket socket,
+                           HttpSession::Configuration config)
+      : config_{config}, stream_(std::move(socket)) {
+    connectOnError([](auto ec, auto &&message) {
+      std::cerr << "http session error " << ec << ": " << message << std::endl;
```

Comment:
You should better use logger instead of iostream, I think

Reply:
fixed